subprojects/kotlin-dsl/src/main/kotlin/org/gradle/kotlin/dsl/accessors/AccessorsClassPath.kt

```diff
@@ -67,62 +87,139 @@ fun projectAccessorsClassPath(project: Project, classPath: ClassPath): Accessors
 
 
 private
-fun buildAccessorsClassPathFor(project: Project, classPath: ClassPath) =
-    configuredProjectSchemaOf(project)?.let { projectSchema ->
+fun buildAccessorsClassPathFor(project: Project, classPath: ClassPath): AccessorsClassPath? {
+    val workExecutor: WorkExecutor<ExecutionRequestContext, CachingResult> = project.serviceOf()
+    val cacheKeyBuilder: CacheKeyBuilder = project.serviceOf()
+    val workspaceProvider: KotlinDslWorkspaceProvider = project.serviceOf()
+    return configuredProjectSchemaOf(project)?.let { projectSchema ->
         // TODO:accessors make cache key computation more efficient
-        cachedAccessorsClassPathFor(project, cacheKeyFor(projectSchema, classPath)) { srcDir, binDir ->
-            withAsynchronousIO(project) {
-                buildAccessorsFor(
-                    projectSchema,
-                    classPath,
-                    srcDir = srcDir,
-                    binDir = binDir
-                )
-            }
+        val cacheKeySpec = cacheKeyFor(projectSchema, classPath)
+        val cacheKey = cacheKeyBuilder.build(cacheKeySpec)
+        workspaceProvider.withWorkspace(cacheKey) { workspace, executionHistoryStore ->
```

Comment:
:o: I think it would be better to have the prefix separated from the cache key, for both readability and symmetry with the plugin accessors generation. But this is currently tangled with the script compilation so it can be done in a follow up PR.

Reply:
Good point, let's do that.